src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts

```diff
@@ -96,23 +86,122 @@ export function handleANSIOutput(text: string, linkDetector: LinkDetector): HTML
+	 * 	a class-defined color.
+	 */
+	function changeColor(newClass: string | number | null, isForeground: boolean, customColor?: RGBColor | null): void {
+		const colorType = isForeground ? 'foreground' : 'background';
+		styleNames = styleNames.filter(style => !style.match(new RegExp(`^code-${colorType}-(\\d+|custom)$`)));
+		if (newClass) {
+			styleNames.push(`code-${colorType}-${newClass}`);
+		}
+		if (isForeground) {
+			customFgColor = customColor || null;
+		} else {
+			customBgColor = customColor || null;
+		}
+	}
+
+	/**
+	 * Calculate and set basic ANSI formatting. Supports bold, italic, underline,
+	 * normal foreground and background colors, and bright foreground and
+	 * background colors. Not to be used for codes containing advanced colors.
+	 * Will ignore invalid codes.
+	 * @param styleCodes Array of ANSI basic styling numbers, which will be
+	 * applied in order. New colors and backgrounds clear old ones; new formatting
+	 * does not.
+	 * @see {@link https://en.wikipedia.org/wiki/ANSI_escape_code }
+	 */
+	function calcAndSetBasicFormatters(styleCodes: number[]): void {
+		for (let code of styleCodes) {
+			if (code === 0) {
+				styleNames = [];
+			} else if (code === 1) {
+				styleNames.push('code-bold');
+			} else if (code === 3) {
+				styleNames.push('code-italic');
+			} else if (code === 4) {
+				styleNames.push('code-underline');
+			} else if ((code >= 30 && code <= 37) || (code >= 90 && code <= 97)) {
+				changeColor(code, true);
+			} else if ((code >= 40 && code <= 47) || (code >= 100 && code <= 107)) {
+				changeColor(code, false);
+			} else if (code === 39) {
+				changeColor(null, true);
+			} else if (code === 49) {
+				changeColor(null, false);
+			}
+		}
+	}
+
+	/**
+	 * Calculate and set styling for complicated 24-bit ANSI color codes.
+	 * @param styleCodes Full list of integer codes that make up the full ANSI
+	 * sequence, including the two defining codes and the three RGB codes.
+	 * @param isForeground If `true`, will set foreground color, if `false`, will
+	 * set background color.
+	 * @see {@link https://en.wikipedia.org/wiki/ANSI_escape_code#24-bit }
+	 */
+	function calcAndSet24BitColor(styleCodes: number[], isForeground: boolean): void {
+		if (styleCodes.length >= 5 &&
+			styleCodes[2] >= 0 && styleCodes[2] <= 255 &&
+			styleCodes[3] >= 0 && styleCodes[3] <= 255 &&
+			styleCodes[4] >= 0 && styleCodes[4] <= 255) {
+			const customColor = new RGBColor(styleCodes[2], styleCodes[3], styleCodes[4]);
+			changeColor('custom', isForeground, customColor);
+		}
+	}
+
+	/**
+	 * Calculate and set styling for advanced 8-bit ANSI color codes.
+	 * @param styleCodes Full list of integer codes that make up the ANSI
+	 * sequence, including the two defining codes and the one color code.
+	 * @param isForeground If `true`, will set foreground color, if `false`, will
+	 * set background color.
+	 * @see {@link https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit }
+	 */
```

Comment:
I do not like the `calcAnd` names. Just use the `set8BitColor`. Remove `calcAnd` from all methods

Reply:
Fixed with 2e17ed2.